daemon/vpn/nordlynx/libtelio/libtelio.go

```diff
@@ -28,40 +28,63 @@ import (
 
 const (
 	// TelioLocalConfigName defines env key for local config value
-	TelioLocalConfigName = "TELIO_LOCAL_CFG"
+	TelioLocalConfigName     = "TELIO_LOCAL_CFG"
+	defaultHeartbeatInterval = 60 * 60
 )
 
 type state struct {
-	Nickname  string `json:"nickname"`
-	State     string `json:"state"`
-	PublicKey string `json:"public_key"`
-	IsVPN     bool   `json:"is_vpn"`
-	IsExit    bool   `json:"is_exit"`
+	Nickname  string
+	State     teliogo.NodeState
+	PublicKey string
+	IsVPN     bool
+	IsExit    bool
 }
 
-type event struct {
-	Body state `json:"body"`
+func maskPublicKey(event string) string {
+	expr := regexp.MustCompile(`"PublicKey":(\s)*"(.*?)"`)
+	return expr.ReplaceAllString(event, `"PublicKey":"***"`)
 }
 
-type eventFn func(string)
+type eventCb func(teliogo.Event) *teliogo.TelioError
 
-func maskPublicKey(event string) string {
-	expr := regexp.MustCompile(`"public_key":(\s)*"(.*?)"`)
-	return expr.ReplaceAllString(event, `"public_key":"***"`)
+func (cb eventCb) Event(payload teliogo.Event) *teliogo.TelioError {
+	return cb(payload)
 }
 
-func eventCallback(states chan<- state) eventFn {
-	return func(s string) {
-		log.Println(internal.InfoPrefix + maskPublicKey(s))
-		var e event
-		if err := json.Unmarshal([]byte(s), &e); err != nil {
-			return
+func eventCallback(states chan<- state) eventCb {
+	return func(e teliogo.Event) *teliogo.TelioError {
+		eventBytes, err := json.Marshal(&e)
+		if err != nil {
+			log.Printf(internal.WarningPrefix+" can't marshal telio Event %T: %s\n", e, err)
+		} else {
+			log.Printf(internal.InfoPrefix+" received event %T: %s\n", e, maskPublicKey(string(eventBytes)))
```

Comment:
Would have been nice if `teliogo.Event` masks the keys automatically.

Reply:
Yes, that's true. 
The `teliogo.Event` is, currently, a super abstract interface. I think it's because of the way the bindings are generated by `uniffi-bingen-go` and it has no idea about its content, it could be tricky to implement this cleanly in bindings